.gitignore

```diff
@@ -165,3 +165,6 @@ dask-worker-space/
 # Sphinx docs & build artifacts
 docs/cudf/source/api_docs/generated/*
 docs/cudf/source/api_docs/api/*
+docs/cudf/source/user_guide/example_files/*
+docs/cudf/source/user_guide/example_output/*
+docs/cudf/source/user_guide/cudf.*Dtype.*.rst
```

Comment:
What does this rule ignore? Are there generated files / where do they come from?

Reply:
There are files generated after build for every attribute & method:
Grayed out ones are the build artifacts
<img width="538" alt="Screen Shot 2022-10-26 at 3 59 09 PM" src="https://user-images.githubusercontent.com/11664259/198136683-6a4034c5-452a-4d58-b9fd-4caf2c70ea6b.png">

Reply:
@galipremsagar Do you know what script generates those files or how that is defined? This doesn't seem like a Sphinx feature that I'm familiar with.

Reply:
That is the expected behavior and happens in all of our sphinx-build. Those files are generated in `docs/cudf/source/api_docs/api/` folder because we specify `toctree: api/`.

The part of code that is responsible for this is everywhere in our code where we do `HACK --...` in comments for example:
```
..
   HACK -- the point here is that we don't want this to appear in the output, but the autosummary should still generate the pages.
   .. autosummary::
      :toctree:
      {% for item in all_methods %}
      {%- if not item.startswith('_') or item in ['__call__'] %}
      {{ name }}.{{ item }}
      {%- endif -%}
      {%- endfor %}
```

This is basically being done to mimic panadas & generate dedicated html pages for each property and function.

Reply:
Thanks for the explanation!